src/components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,53 @@
+import { Link } from "@canonical/react-components";
+import { useParams } from "react-router-dom";
+
+export type EntityDetailsRoute = {
+  userName: string;
+  modelName: string;
+  appName: string;
+};
+
+const Breadcrumb = (): JSX.Element => {
+  const { userName, modelName, appName } = useParams<EntityDetailsRoute>();
+
+  return (
+    <nav className="p-breadcrumbs" aria-label="Breadcrumb navigation">
+      <ol className="p-breadcrumbs__items">
+        {appName ? (
+          <>
+            <li className="p-breadcrumbs__item" data-test="breadcrumb-model">
+              <Link
+                href={`/models/${
+                  userName ? `${userName}/${modelName}` : `${modelName}`
```

Comment:
This logic is a bit hard to follow and repeated, can you abstract it out into a utility function in this file?

Reply:
Good idea